src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -6874,6 +6898,38 @@ void ClassFileParser::parse_stream(const ClassFileStream* const stream,
   // all bytes in stream read and parsed
 }
 
+// Return true if the specified class is not a valid super class for an inline type.
+// A valid super class for an inline type is abstract, has no instance fields,
+// does not implement interface java.lang.IdentityObject (checked elsewhere), has
+// an empty body-less no-arg constructor, and no synchronized instance methods.
```

Comment:
The comment should be more explicit about the fact that it is performing a local test, without considering super types. The final decision taking account super types being performed in `fill_instance_klass()`.

Reply:
The latest change updates this comment.